object/state.go

```diff
@@ -28,12 +39,58 @@ func (e *Environment) Get(name string) (Object, bool) {
 	return e.outer.Get(name) // recurse.
 }
 
-func (e *Environment) Set(name string, val Object) Object {
+// Defines constant as all CAPS (with _ ok in the middle) identifiers.
+// Note that we use []byte as all identifiers are ASCII.
+func Constant(name string) bool {
+	for i, v := range name {
+		if i != 0 && v == '_' {
+			continue
+		}
+		if v < 'A' || v > 'Z' {
```

Comment:
Constant may/could also contains number, no?

Reply:
well... not right now indeed :) PI2=PI; PI2++ works https://grol.io/?c=PI2=PI+PI2%2B%2BPI2